design/specs/listing-views.md

```diff
@@ -1,27 +1,109 @@
+title: Listing Views Spec
 description: 
 published: true
-date: 2021-09-17T08:38:09.720Z
+date: 2021-12-20T09:00:00.201Z
 tags: 
 editor: markdown
-dateCreated: 2021-09-17T08:28:58.527Z
+dateCreated: 2021-12-20T09:00:00.201Z
 ---
 
-# Context
-When using Mathesar, users should rely on views for most analysis tasks and tables for more structural tasks like modeling and data transformation. However, both objects are similar and share some functionality. A clear distinction needs to be made between tables and views. Views should also take a more prominent space, and users in a table context should understand that saving as a view will allow them to choose a subset of columns or combine with other tables. 
+## Context
 
-# Prototype
-[Prototype for Listing and Finding Views](https://www.figma.com/proto/Uaf1ntcldzK2U41Jhw6vS2/Mathesar-MVP?page-id=5343%3A76906&node-id=5343%3A76907&viewport=338%2C48%2C0.72&scaling=contain&starting-point-node-id=5343%3A76907)
+Database objects are defined objects in a database that store or reference data. In the context of Mathesar, two of those objects are listed and accessible to users as `Tables` and `Views`. Together, these objects are at the core of Mathesar's user experience. One part of that experience is finding and opening these objects and differentiating both.
+
+## Prototype
+
+### Link
+
+[Prototype](https://mathesar-prototype.netlify.app/)
+
+### Videos
+
+- Scenario 1: [Video](https://www.loom.com/share/179070149f00419e9125bd8fba675aa4)
+- Scenario 2: [Video](https://www.loom.com/share/e32500bc744246e5b926852231577fd7)
+- Scenario 3: [Video](https://www.loom.com/share/d553e7edbb28481b8aed02b055270154)
+- Scenario 4: [Video](https://www.loom.com/share/91a20c4770ef4439860425a1bcd84db6)
+
+## User Experience
+
+### Listing of Views and Searching by Name
+
+With the introduction of `Views`, the user interface needs to be updated to retrieve and identify all objects easily. `Views` in the same manner as `Tables`, have names that allow users to identify them or search them.
+
+Initially, users will only be able to search views by name. However, if a user hasn't encountered a particular object very often, relying on a name alone might not be sufficient to ensure a good user experience.
 
-# User Experience
 ## Scenarios
 
-### User sees a list of all views and can filter or sort the list items
-A user can see a list of all views for a schema in the sidebar area. They can filter the list using the search input or sort them by alphabetical order or last modified. Last modified is the default sorting for the list of views.
+### Scenario 1: User browses through a list of all available views in a currently open schema
+
+#### Scenario 1a: From the schema details view
+
+- User navigates to the schema details page by navigating directly to a URL that points to this page or by clicking on the schema link at the top navigation bar.
+- User sees a list of all tables and views for the current schema.
```

Comment:
Views and Tables are going to have pretty different functionality so I think we should separate them into two different sections on the schema detail page (rather than showing them as the same list) to make the distinction clearer to the user.

I do think we can have a unified search for both, although maybe we'll still want to split search results into sections.

Reply:
It might also be a good idea to show some metadata about the tables and views. e.g. we can show the number of records or rows, and maybe some details about the column names? It would help users identify what they're looking for if they don't have great names for their DB objects.

Reply:
@kgodey I had them separated initially for the same reason, but as I was thinking about the workflow I imagined a user would be jumping from view to table, to view again especially if they are creating a view from an existing table and I liked the idea of having them as a unified view. Maybe this could be solved by tabs. I'll consider this and make changes.

Reply:
I think there's other ways to solve users being able to jump between tables and views easily. It's really important for users to be able to tell the difference between Views and Tables.